server.js

```diff
@@ -209,14 +212,77 @@ app.get('/leaderboard-table', function(req, res) {
 
     ft = true;
     shrt = false;
+    
+    if (req.query.fullTable == 'false')
+        ft = false;
+    if (req.query.longTable == 'false')
+        shrt = true;
+    
+    if(req.query.type === 'points'){
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = pointsTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'overall') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = leaderboardTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'accuracy') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = accuracyTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    } else if (req.query.type === 'attempt') {
+        lb.leaderboard(req.session.user.id, shrt, req.query.type, function(leader) {
+            var html = attemptTable({
+                fullTable: ft,
+                shortTable: shrt,
+                leaderboard: leader,
+                userid: req.session.user.id
+            });
+    
+            return res.status(200).send(html);
+        });
+    }
+});
 
+/* Fetch and render the points table. Send HTML as response. */
+app.get('/points-leaderboard', function(req, res) {
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    var ft, shrt;
```

Comment:
why do you need to define these variables then assign them? you can just do:
``` javascript
var variableName = true;
```

Reply:
This was there from before, I'll change it though.